discovery_child_development/pipeline/labelling/detection_management/detection_management_labels.py

```diff
@@ -0,0 +1,197 @@
+        )
+        labelled = load_jsonl(str(OUTPUT_FILEPATH / output_filename) + ".jsonl")
+        labelled_ids = pd.DataFrame(labelled).id.to_list()
+        texts_df = texts_df[~texts_df.id.isin(labelled_ids)]
+    except:
+        logging.info("No labelled data found")  # noqa: T001
+    # Subsample
+    if (num_samples == 0) or (num_samples > len(texts_df)):
+        num_samples = len(texts_df)
+    texts_df = texts_df.sample(num_samples).reset_index(drop=True)
+
+    # Fetch the categories
+    categories = json.loads(PATH_TO_CATEGORIES.read_text())
+
+    logging.info(f"Number of texts to classify: {len(texts_df)}")  # noqa: T001
+    logging.info(f"Number of distinct categories: {len(categories)}")  # noqa: T001
+
+    # Load instruction messages
+    messages = json.load(open(PATH_TO_MESSAGE_PROMPT, "r"))
+    messages = [MessageTemplate.load(m) for m in messages]
+    function = FunctionTemplate.load(str(PATH_TO_FUNCTION))
+    examples = create_examples_string(load_jsonl(PATH_TO_EXAMPLES))
+
+    for i, batched_results in enumerate(batch(texts_df, 10)):
+        logging.info(f"Batch {i} / {len(texts_df) // 10}")  # noqa: T001
+        tasks = [
+            Classifier.agenerate(
+                model=model,
+                temperature=temperature,
+                messages=messages,
+                message_kwargs={
+                    "category_description": create_category_description_string(
+                        categories, randomise=True
+                    ),
+                    "n_categories": len(categories),
+                    "examples": examples,
+                    "text": tup.text,
+                    "id": tup.id,
+                    "source": tup.source,
+                },
+                functions=[function.to_prompt()],
+                function_call={"name": "predict_detection_management"},
+                max_tokens=100,
+                concurrency=5,
+            )
+            for tup in batched_results.itertuples()
+        ]
+
+        for future in asyncio.as_completed(tasks):
+            result = await future  # Get the result (waits if not ready)
+            # Write to the file
+            await Classifier.write_line_to_file(
+                result,
+                OUTPUT_FILEPATH,
+                output_filename,
+            )
+
+        time.sleep(10)
+
+    # Upload to S3
+    S3.upload_file(
+        path_from=str(OUTPUT_FILEPATH / output_filename) + ".jsonl",
+        bucket=S3_BUCKET,
+        path_to=s3_path,
+    )
+
+
+def parse_arguments():
+    """Parse the arguments passed to the script."""
+    # Create the parser
+    parser = argparse.ArgumentParser(description="Process the arguments for the script")
+    # Add the arguments
+    parser.add_argument("--dataset", type=str, help="The dataset")
+    parser.add_argument("--model", type=str, help="The model")
+    parser.add_argument("--num_samples", type=int, help="The number of samples")
```

Comment:
Do you not want any defaults for any of the arguments? Or will it change every time?